tests/solvers/test_sundials.py

```diff
@@ -1085,6 +1103,31 @@ def handle_event(self,solver, event_info):
         sim.simulate(2.5)
         
         nose.tools.assert_almost_equal(sim.t, 2.000000, 4)
+
+    @testattr(stddist = True)
+    def test_terminate_simulation_external_event(self):
+        """
+        This tests the functionality of raising TerminateSimulation exception in handle_result. External event detection.
+        """
+        class Extended_Problem(Implicit_Problem):
+            def __init__(self):
+                pass
+            def handle_event(self,solver, event_info):
+                if solver.t > 1.5:
+                    raise TerminateSimulation
+            res = lambda self,t,y,yd,sw: np.array([y[0]-1.0])
+            state_events = lambda self,t,y,yd,sw: np.array([t-1.0, t-2.0])
+            y0 = [1.0]
+            yd0 = [1.0]
+            sw0 = [False]
+
+        prob = Extended_Problem()
+    
+        sim = IDA(prob)
+        sim.external_event_detection = True
+        sim.simulate(2.5)
+        
+        nose.tools.assert_almost_equal(sim.t, 2.000000, 4)
```

Comment:
Why do test that the simulation time is almost equal to `2` if we terminate the simulation at `t>1.5`? Is it because we took `t` changed from `1.5` to `2.0` just before the simulation ended or something similar?

Reply:
The t > 1.5 is in the _handle_event_ function and is only called during events. The _state_events_ function defines the conditions for events, here:

t-1.0, t-2.0, i.e., events at t = 1 and t = 2, only the second one leads to termination.